doc/design/draft/named-segments.md

```diff
@@ -0,0 +1,228 @@
+
+```
+[general]
+version = 2
+
+[[segment]]
+name = "foo"
+
+[[segment.mempool]]
+size = 32
+count = 10000
+```
+
+Updating the version will be necessary for older versions of RouDi to fail informatively when presented with a newer config. Otherwise RouDi can support version 1 configs as if they are version 2 configs with no `name` fields specified.
+
+### Requesting Segments
+
+#### When creating a Publisher
+
+A new field will be added to the [PublisherOptions struct](../../../iceoryx_posh/include/iceoryx_posh/popo/publisher_options.hpp) as follows:
+
+```
+struct PublisherOptions
+{
+  ...
+  ShmName_t segmentName{""};
+```
+
+#### When creating a Subscriber
+
+A similar field will be added to the [SubscriberOptions struct](../../../iceoryx_posh/include/iceoryx_posh/popo/subscriber_options.hpp), except that it will support multiple elements
+
+```
+struct SubscriberOptions
+{
+  ...
+  vector<ShmName_t, MAX_SUBSCRIBER_SEGMENTS> segmentNames{};
+```
+
+`MAX_SUBSCRIBER_SEGMENTS` can be set to some reasonably small value to start with since explicitly allowing many but not all read-access segments is an unlikely use case. 
+
+If this assumption turns out to be wrong however, we can always update it to be `MAX_SHM_SEGMENTS` instead.
+
+#### When creating Clients and Servers
+
+Clients and Servers will have similar fields, distinguished by request/response:
+
+```
+struct ClientOptions
+{
+  ...
+  ShmName_t requestSegmentName{""};
+  vector<ShmName_t, MAX_RESPONSE_SEGMENTS> responseSegmentNames{};
+```
+
+```
+struct ServerOptions
+{
+  ...
+  vector<ShmName_t, MAX_REQUEST_SEGMENTS> requestSegmentNames{};
+  ShmName_t responseSegmentName{""};
+```
+
+#### When initializing the runtime
+
+We may wish to also additionally support requesting segments upon runtime initialization. The motivation here is that if we know we will only need to use certain segments, we can avoid expending unecessary resources mapping the ones we will not use. Additionally, this can be used as an error mitigation technique to prevent communication endpoints from requesting an unsupported segment.
+
+Currently initialization of the runtime happens in the constructor, and any errors that occur result in a contract violation and program termination. 
+
+While it is not directly in scope of this design, it would be beneficial to refactor the Runtime implementations to use the builder pattern such that any errors may be handled through returning errors types.
+
+Tying this together with requesting segments, this could look something like:
+```
+class DefaultRuntimeBuilder
+{
+  ...
+  IOX_BUILDER_PARAMETER(vector<ShmName_t>, shmSegmentNames, {});
+  IOX_BUILDER_PARAMETER(bool, allowUnmappedSegments, false);
+  ...
+  public:
+    expected<DefaultRuntime, RuntimeBuilderError> create() noexcept;
+```
+
+The `create` method of this builder would then return a custom error if a 
+segment was requested that either does not exist or the current process does not have access to.
```

Comment:
I would decouple the creation of the runtime from the mapping of the segments. This makes it easier to continue execution and react on unavailable segments than just aborting the creation of the runtime.

Reply:
How about adding a flag indicating whether none, all read-only, all write, or all read and write segments should be mapped upon creation? And then additionally the ability to map specific ones? 

We can of course also expose those as separate functions as well.

Reply:
I think that might work.